### PR TITLE
fix(Portal): update `mountNode` types to accept HTMLElement

### PR DESCRIPTION
**Pull request by layershifter**
## Current Behavior

`mountNode` accepts only `HTMLDivElement`

## New Behavior

`mountNode` accepts only `HTMLElement`

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox</abbr>                                                                              |    `6.817 kB`<br />`2.901 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `176.084 kB`<br />`49.112 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `32.526 kB`<br />`10.645 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `105.847 kB`<br />`32.431 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `109.026 kB`<br />`32.895 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 |  `96.782 kB`<br />`29.558 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |    `6.267 kB`<br />`2.168 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `42.789 kB`<br />`14.712 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/8bbebba6e62f3cdb1876239cf1511a412aa2519c'>8bbebba6e62f3cdb1876239cf1511a412aa2519c</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-mount_base.html">271</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-mount.html">327</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/Avatar-mount_base.html">904</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/Avatar-mount.html">879</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/Button-mount_base.html">539</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/Button-mount.html">533</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProvider-mount_base.html">1865</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProvider-mount.html">1926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-mount_base.html">271</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-mount.html">327</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">210</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">222</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">289</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">287</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/MakeStyles-mount_base.html">1583</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22153/perf-test-react-components/MakeStyles-mount.html">1560</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by ling1726** [comment]
do we actually have a requirement for this ?

**Comment by layershifter** [comment]
> do we actually have a requirement for this ?

No, but I think that it's initially wrong implementation.

**Review by spmonahan — APPROVED** [review]

File: packages/react-portal/src/components/Portal/Portal.types.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ type PortalCommons = {
    * Where the portal children are mounted on DOM
    * @defaultValue a new element on document.body without any styling
    */
-  mountNode: HTMLDivElement | undefined;
+  mountNode: HTMLElement | undefined;
```
Comment (micahgodbolt) [inline packages/react-portal/src/components/Portal/Portal.types.ts:12]:
```suggestion
  mountNode: HTMLElement | null;
```

this undefined should be null

**Review by micahgodbolt — APPROVED** [review]

**Pull request merged**